### PR TITLE
CI: rebalance the travis ci jobs

### PR DESCRIPTION
**Pull request by jreback**
move some from the 3.6 build to 3.7 as the 3.6 is 15 min longer. mark the multi-thread tests as slow.

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

File: pandas/tests/test_resample.py [hunk 1]
```diff
@@ -650,6 +650,7 @@ def test_asfreq_fill_value(self):
     def test_resample_interpolate(self):
         # # 12925
         df = self.create_series().to_frame('value')
+        df = df.iloc[0:(len(df) // 2)]
```
Comment (TomAugspurger) [inline pandas/tests/test_resample.py]: I'm curious, why this change?

Comment (jreback) [reply pandas/tests/test_resample.py]: oh that test appears as the number 1 duration (like 30 s), locally it took 10 though, so made it 1/2

**Comment by jreback** [comment]
is there a way to skip doc tests if something is not installed? e.g. https://travis-ci.org/pandas-dev/pandas/jobs/446665579

I want to run the doc-tests now on both jobs but skip when the deps are not there

**Comment by TomAugspurger** [comment]
I'm not aware of a way to skip doctests based on imports.

File: .travis.yml [hunk 2]
```diff
@@ -52,10 +52,10 @@ matrix:
           - python-gtk2
     - dist: trusty
       env:
-        - JOB="3.6, lint, coverage" ENV_FILE="ci/travis-36.yaml" TEST_ARGS="--skip-slow --skip-network" PANDAS_TESTING_MODE="deprecate" COVERAGE=true LINT=true
+        - JOB="3.6, lint, coverage" ENV_FILE="ci/travis-36.yaml" TEST_ARGS="--skip-slow --skip-network" PANDAS_TESTING_MODE="deprecate" COVERAGE=true LINT="lint|patterns"
```
Comment (datapythonista) [inline .travis.yml]: I don't think this is working as expected. `LINT` is only checked to be set, whatever is the value, would be considered as `true`. The changes in `code_checks.py` act on the `CHECK` environment variable, which is the parameter that the script receives (and different from this `LINT`.

Now the whole lint is being executed in both jobs.

I'd personally wait until #22854. We can see after the work there whether something like this still makes sense.

File: ci/code_checks.sh [hunk 3]
```diff
@@ -24,7 +24,7 @@ CHECK=$1
 
 
 ### LINTING ###
-if [[ -z "$CHECK" || "$CHECK" == "lint" ]]; then
+if [[ -z "$CHECK" || "$CHECK/lint" == "lint" ]]; then
```
Comment (datapythonista) [inline ci/code_checks.sh]: I may be missing something, but I think the equality with this will compare `"/lint" == "lint"` or `"lint/lint" == "lint"` and will be always false.

**Review by datapythonista — CHANGES_REQUESTED** [review]
I think the changes to linting are not working as expected.

Needs a rebase.

**Comment by jreback** [comment]
yeah i will rebase this later

**Review by datapythonista — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
its ok as is now
